### PR TITLE
ssh-agent-filter 0.4.2 (new formula)

### PR DESCRIPTION
**Pull request by kouk**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [ ] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

```
ssh-agent-filter:
  * GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)
```

However this [package is in debian stable](https://packages.debian.org/jessie/net/ssh-agent-filter)  since [2013](https://tracker.debian.org/pkg/ssh-agent-filter) and it's [popularity is growing](https://qa.debian.org/popcon.php?package=ssh-agent-filter).

### CODE REVIEW

File: Formula/ssh-agent-filter.rb
Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: Can you please submit this patch to the upstream developers of this project as a pull request and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

File: Formula/ssh-agent-filter.rb
Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: Can you please submit this as an issue to the upstream developers of this project and add a link to the upstream issue submission in a comment in the formula file. Thanks!

File: Formula/ssh-agent-filter.rb
Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: Is there no `make install` we can use?

File: Formula/ssh-agent-filter.rb
Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: Remove all these comments.

File: Formula/ssh-agent-filter.rb
Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by MikeMcQuaid** [comment]
> ssh-agent-filter:
>   * GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)
> However this package is in debian stable since 2013 and it's popularity is growing.

I'm ok with accepting this project when it at least the `master` branch works without patching on macOS.

File: Formula/ssh-agent-filter.rb
Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: Adding one upstream would be amazing if you felt like submitting a patch there?

File: Formula/ssh-agent-filter.rb
Comment (kouk) [inline Formula/ssh-agent-filter.rb]: @MikeMcQuaid good idea, I pushed a better test.

File: Formula/ssh-agent-filter.rb
Comment (kouk) [inline Formula/ssh-agent-filter.rb]: @MikeMcQuaid For the first chunk I added it as an external patch. For the second I will comment below.

File: Formula/ssh-agent-filter.rb
Comment (kouk) [inline Formula/ssh-agent-filter.rb]: I had already submitted the first part of the patch here: https://github.com/tiwe-de/ssh-agent-filter/pull/5
I added it to the formula as an external patch.

**Comment by kouk** [comment]
@MikeMcQuaid one thing I'm not sure about is a missing dependency of help2man, the perl library Locale::gettext, which I worked around by removing the relevant command line arguments from the Makefile. Is there a package in Homebrew that contains this perl module? Otherwise the master branch will not build the man page successfully, and I think this package is installed by default in Debian.

File: Formula/ssh-agent-filter.rb
Comment (kouk) [inline Formula/ssh-agent-filter.rb]: done

File: Formula/ssh-agent-filter.rb
Comment (kouk) [inline Formula/ssh-agent-filter.rb]: @MikeMcQuaid this wasn't actually necessary in the end so I removed it.

File: Formula/ssh-agent-filter.rb
Comment (tiwe-de) [inline Formula/ssh-agent-filter.rb]: This needs considerable processing time for finding primes. You might use `-t ed25519` to select a key type that is much faster to generate. In that case you should also use `-C keyA` to set the comment or ssh-add/ssh-agent might use the embedded comment instead of the filename.

File: Formula/ssh-agent-filter.rb
Comment (tiwe-de) [inline Formula/ssh-agent-filter.rb]: `mydir` appears to be unused

File: Formula/ssh-agent-filter.rb
Comment (tiwe-de) [inline Formula/ssh-agent-filter.rb]: `==` is a bashism, use `=` instead

**Review by tiwe-de — CHANGES_REQUESTED** [review]

File: Formula/ssh-agent-filter.rb
Comment (tiwe-de) [inline Formula/ssh-agent-filter.rb]: It seems like you only want to install some of the files the Debian package contains. The missing parts are at least:
- afssh.1 (needs pandoc to generate it)
- ssh-agent-filter.bash-completion (which version of bash-completion is available on OS X?)

**Comment by tiwe-de** [comment]
the removed option to `help2man` seems to be unnecessary  on OS X or with that version of `help2man`. On a GNU system with full-blown `help2man` it is needed for it to not mangle characters outside of ASCII (in my name).

**Comment by kouk** [comment]
@tiwe-de Thanks for your suggestions, I pushed the changes.

File: Formula/ssh-agent-filter.rb
Comment (kouk) [inline Formula/ssh-agent-filter.rb]: @tiwe-de sorry, I'm not sure I understand what that is. Could you give me a pointer?

Comment (tiwe-de) [inline Formula/ssh-agent-filter.rb]: This should read "pre-generated", sorry.

Comment (kouk) [inline Formula/ssh-agent-filter.rb]: @tiwe-de if you know of a nice way to get the pre-generated keys in the Formula I would do that. But having them verbatim inside the file seems a bit overkill.

File: Formula/ssh-agent-filter.rb
Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

File: Formula/ssh-agent-filter.rb
Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: I've commented there but we need to figure out how to get something merged there or can't accept this (as it shows upstream doesn't want to support macOS).

File: Formula/ssh-agent-filter.rb
Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: You probably want to add `set -e` and perhaps also `set -x` in here.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/ssh-agent-filter.rb
Comment (kouk) [inline Formula/ssh-agent-filter.rb]: @tiwe-de I hesitated to add a dependency on pandoc because that means installing GHC. If there's a way in homebrew to remove a build dependency after then that might not hurt.

Comment (tiwe-de) [inline Formula/ssh-agent-filter.rb]: Isn't that what `=> :build` is for? So a user who installs the binary package ("bottle") would not need to install it.

File: Formula/ssh-agent-filter.rb
Comment (kouk) [inline Formula/ssh-agent-filter.rb]: @MikeMcQuaid `set -e` for sure, but not sure why `set -x` is of benefit.

Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: It will make it easier to debug a failure.

Comment (tiwe-de) [inline Formula/ssh-agent-filter.rb]: if you use `-e` you might want to use a `trap` to clean up.

Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: `test do` already runs in a (cleaned-up) temporary directory so the `mktemp -d` is probably unnecessary.

File: Formula/ssh-agent-filter.rb
Comment (tiwe-de) [inline Formula/ssh-agent-filter.rb]: Actually the manpage contained in the bottles generated by jenkins have the correct umlaut.
Can you test if the output manpage changes with the Locale::gettext perl module installed?

File: Formula/ssh-agent-filter.rb
Comment (kouk) [inline Formula/ssh-agent-filter.rb]: @MikeMcQuaid done.

File: Formula/ssh-agent-filter.rb
Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: Use `https://github.com/tiwe-de/ssh-agent-filter/pull/6.diff` as the URL to remove the need for the comment.

File: Formula/ssh-agent-filter.rb
Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: Again, this probably wants `set -e` and `set -x`.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/ssh-agent-filter.rb
Comment (MikeMcQuaid) [inline Formula/ssh-agent-filter.rb]: Please add a one-line description of what this pull request is on the line above. It also may be worth waiting until this is merged and confirmed to be correct.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by tiwe-de** [comment]
it seems you are trying to enable locale/gettext support in help2man at runtime, but it is disabled at "compile" time; see [the source of help2man](https://sources.debian.net/src/help2man/1.47.4/help2man.PL/).

If help2man in homebrew is always "compiled" without gettext support, you can just leave out the locale setting because non-ascii characters will not be mangled then.

**Comment by MikeMcQuaid** [comment]
@kouk Any news on the comments? Thanks!

**Comment by kouk** [comment]
@MikeMcQuaid sorry, I didn't have time to spare the last two weeks. I will try and work on it in the coming days, but if I understand @tiwe-de 's comment at least one patch will be required in the formula due to help2man not having gettext support in MacOS.

**Comment by MikeMcQuaid** [comment]
@kouk That's ok, just checking in. As long as the patches are submitted upstream (and given how responsive they are ideally merged) before we merge this. Thanks again!

**Comment by fxcoudert** [comment]
Low-visibility github repo, still does not work unpatched on macOS, no commits since October 2016. Upstream needs to move before we can consider adding this formula.

**Pull request closed without merging**
